common/djangoapps/student/views.py

```diff
@@ -1782,7 +1782,7 @@ def _do_create_account(form, custom_form=None):
 
     profile_fields = [
         "name", "level_of_education", "gender", "mailing_address", "city", "country", "goals",
-        "year_of_birth"
+        "year_of_birth", "profession", "specialty"
```

Comment:
Actually, I think all you have to do is remove "profession and "speciality" from this list and those fields should get persisted to UserProfile.meta.